gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/util/detached/DetachedFactory.java

```diff
@@ -115,4 +116,74 @@ public static <D> D detach(final Object object, final boolean withProperties) {
             return (D) object;
         }
     }
+
+    // detached with options
+    public static DetachedVertex detach(final Vertex vertex, final DetachStrategy.DetachOptions detachOptions) {
```

Comment:
i think we'd want to deprecate the `withProperties` overloads for these methods and prefer this new one. thoughts?

Reply:
This overloads used in some other cases like Gryo Serializers, `DedupGlobalStep`, `EventStrategy`, etc.
So it seems to me that it is more convenient to keep `withProperties` as shortcut